lib/services/summarizer.go

```diff
@@ -98,6 +101,34 @@ type Summarizer interface {
 	AllInferencePolicies(ctx context.Context) iter.Seq2[*summarizerv1.InferencePolicy, error]
 }
 
+// inferencePolicyValidationContext is a special kind of [RuleContext] that
+// allows validating inference policy filters by attempting to resolve
+// identifiers against other supported resource types.
+type inferencePolicyValidationContext struct {
+	Context
```

Comment:
I would rather build a new context than embedding the existing one., specially because it only supports 3 fields, resource, user and you can use it in the e part of the code for the filter to properly work